### DISCUSSION TITLE
自定义容器如何定制代码块的名字

### DISCUSSION THREAD

**Discussion by jdysya** [vuepress-theme-hope/vuepress-theme-hope#3056] [Ideas]
由于平时记笔记比较偏向于`obsidian`，里面有一款插件，实现效果和`vuepress`中的自定义容器的效果是类似的，不同之处在于，比如`obsidian`中是这样的:
````
```ad-tip
// 内容
```
````
而`vuepress`中采用的是`markdown-it`中的格式：
```
:::tip
// 内容
:::
```
我看文档中[样式化](https://vuepress-theme-hope.github.io/v2/zh/guide/markdown/stylize.html#%E4%BD%BF%E7%94%A8)有提到创建行内 snippet，对内联标记进行样式化，不知道这种方法能否实现将`obsidian`中的容器转换为`vuepress`中的容器呢

**Comment by Mister-Hope** [comment]
你可以把obsidian支持的语法都列出来，我们可以支持。

**Reply by jdysya** [reply]
`obsidian`中的容器是这个[插件实现的](https://github.com/valentine195/obsidian-admonition),新版的`obsidian`内置了容器,[可以参考这个文档](https://help.obsidian.md/Editing+and+formatting/Callouts)，我觉得还是这种代码块风格的比较好，我常用的几种类型就是`note`,`tip`,`summary`,`warning`，而且`obsidian`也有比较丰富的插件生态，不知道`vuepress`能否开发插件让`obsidina`渲染[vuepress-plugin-md-enhance](https://vuepress-theme-hope.github.io/v2/md-enhance/zh/)中的语法呢，我觉得`obsidian`对文档的编辑体验是优于`vscode`的

**Reply by jdysya** [reply]
1. 首先`obsidian`支持实时渲染markdown的方式
2. `obsidian`中对其他文件的引用的链接是可以实时更新的，比如我在`a.md`中引用了同路径下的`b.md`，当我将`b.md`移动到 其他文件夹后，`obsidian`会自动更新这个引用的链接
3. `obsidan`的插件生态开发语言和`vuepress`也是一致的，应该在某些方面可以复用
